### PR TITLE
Refactor AnnotationServerStorage as a class

### PR DESCRIPTION
**Pull request by rlskoeser**

### CODE REVIEW

File: tests/index.test.ts [hunk 1]
```diff
@@ -150,4 +151,48 @@ describe("Event handlers", () => {
...
+            "@context": "oldfakeContext",
+            id: "oldId",
+            body: {},
+            motivation: "commenting",
+            target: { source: "oldfakesource" },
+            type: "Annotation",
+        };
+
+        clientMock.emit("updateAnnotation", annotation, previous);
+        // should call AdjustTargetSource
```
Comment (rlskoeser) [inline tests/index.test.ts:174]: @blms I can't figure out how to emit with two params. According to the tin-emitter docs I think this syntax is correct, but when the handler is called the second parameter is `undefined`. Help?

Comment (blms) [reply tests/index.test.ts:174]: Ah, I see the problem! It's an issue with my mock. I'll adjust the mock to work properly with multiple arguments. (I mocked it so I didn't have to rely on importing tiny-emitter itself and deal with any potential problems there)

Comment (blms) [reply tests/index.test.ts:174]: Should be fixed now

Comment (rlskoeser) [reply tests/index.test.ts:174]: I didn't even think to look in the local code! Thanks, would not have found that

**Review by blms — COMMENTED** [review]
This looks great! Especially appreciate the initial adapter tests.

**Pull request merged**